src/bosh_azure_cpi/lib/cloud/azure/models/disk_id.rb

```diff
@@ -26,103 +25,76 @@ class DiskId
+      obj_id.validate
+      obj_id
     end
 
     def disk_name
-      return @id if @version == VERSION1
-      @id['disk_name']
+      return @plain_id unless @plain_id.nil?
+      @id_hash[DISK_NAME_KEY]
     end
 
     def caching
-      cloud_error('This function should only be called for data disks') unless disk_name.start_with?(DATA_DISK_PREFIX, MANAGED_DATA_DISK_PREFIX)
+      invalid = !disk_name.start_with?(DATA_DISK_PREFIX, MANAGED_DATA_DISK_PREFIX)
+      cloud_error('This function should only be called for data disks') if invalid
 
-      return parse_data_disk_caching_v1(@id) if @version == VERSION1
-      @id['caching']
+      return _parse_data_disk_caching_plain(@plain_id) unless @plain_id.nil?
+      @id_hash[CACHING_KEY]
     end
 
     def storage_account_name
-      cloud_error('This function should only be called for unmanaged disks') if disk_name.start_with?(MANAGED_DATA_DISK_PREFIX)
+      invalid = disk_name.start_with?(MANAGED_DATA_DISK_PREFIX)
+      cloud_error('This function should only be called for unmanaged disks') if invalid
 
-      return parse_storage_account_v1(@id) if @version == VERSION1
-      @id['storage_account_name']
+      return _parse_storage_account_plain(@plain_id) unless @plain_id.nil?
+      @id_hash[STORAGE_ACCOUNT_NAME_KEY]
     end
 
     def validate
-      if @version == VERSION2
-        cloud_error("Invalid disk_name in disk id (version 2) '#{@id}'") if disk_name.nil? || disk_name.empty?
-        cloud_error("Invalid caching in disk id (version 2) '#{@id}'") if caching.nil? || caching.empty?
+      if @plain_id.nil?
+        cloud_error("Invalid disk_name in disk id (version 2) '#{self}'") if disk_name.nil? || disk_name.empty?
+        cloud_error("Invalid caching in disk id (version 2) '#{self}'") if caching.nil? || caching.empty?
         unless resource_group_name.nil?
-          cloud_error("Invalid resource_group_name in disk id (version 2) '#{@id}'") if resource_group_name.empty?
+          cloud_error("Invalid resource_group_name in disk id (version 2) '#{self}'") if resource_group_name.empty?
         end
         if disk_name.start_with?(DATA_DISK_PREFIX)
-          cloud_error("Invalid storage_account_name in disk id (version 2) '#{@id}'") if storage_account_name.nil? || storage_account_name.empty?
+          cloud_error("Invalid storage_account_name in disk id (version 2) '#{self}'") if storage_account_name.nil? || storage_account_name.empty?
         end
       end
     end
 
+    private_class_method def self._generate_data_disk_name(use_managed_disks)
+      "#{use_managed_disks ? MANAGED_DATA_DISK_PREFIX : DATA_DISK_PREFIX}-#{SecureRandom.uuid}"
+    end
+
     private
 
-    def parse_data_disk_caching_v1(disk_name)
+    def _parse_data_disk_caching_plain(disk_name)
       caching = disk_name.split('-')[-1]
       validate_disk_caching(caching)
       caching
     end
 
-    def parse_storage_account_v1(disk_name)
+    def _parse_storage_account_plain(disk_name)
```

Comment:
_function_name is mostly likely python style.... not ruby.

we better to follow same code style as much as possible. If you change the style here, please also change private methods of other classes to follow the same style.

Reply:
you are right.
doing it in this branch https://github.com/andyliuliming/bosh-azure-cpi-release/tree/develop_private_methods